akka-cluster-sharding/src/main/scala/akka/cluster/sharding/ShardCoordinator.scala

```diff
@@ -801,11 +801,25 @@ class PersistentShardCoordinator(typeName: String, settings: ClusterShardingSett
   }: Receive).orElse[Any, Unit](receiveTerminated).orElse[Any, Unit](receiveSnapshotResult)
 
   def receiveSnapshotResult: Receive = {
-    case SaveSnapshotSuccess(_) ⇒
+    case SaveSnapshotSuccess(m) ⇒
       log.debug("Persistent snapshot saved successfully")
+      deleteMessages(m.sequenceNr)
```

Comment:
I would like to keep at least one "batch" (the `snapshotAfter` count) of events around because
* it's not safe to delete all events immediate because snapshots are typically stored with a weaker consistency level which means that a replay might "see" the deleted events before it sees the stored snapshot, i.e. it will use an older snapshot and then not replay the full sequence of events
* for debugging if something goes wrong in production it's very useful to be able to inspect the events

I think something like this:
```
// delete old events but keep the latest around because... (add my comments above)
val deleteToSequenceNr = m.sequenceNr - 2 * snapshotAfter
if (deleteToSequenceNr > 0)
  deleteMessages(deleteToSequenceNr)
```